### DISCUSSION TITLE
Support for selective throttling(blocking specific user after first request for specific time until first one is served)

### DISCUSSION THREAD

**Discussion by SasikiranJ** [fastapi/fastapi#7134] [Questions] [answered]

**Comment by coder3112** [comment, marked as answer]
I think you can add a custom middle ware with custom code.
For eg:
``` python
import time

from fastapi import FastAPI, Request

app = FastAPI()

class SomeError(Exception):
    # Wite your exception
     pass

@app.middleware("http")
async def add_process_time_header(request: Request, call_next):
    start_time = time.time()
    response = await call_next(request)
    process_time = time.time() - start_time
    if process_time < 1000:
        raise SomeError
    return response

**Comment by SasikiranJ** [comment]
Thank you for your reply

**Comment by Mause** [comment]
Is there a reason why this library doesn't suite your needs?

https://github.com/abersheeran/asgi-ratelimit

**Comment by SasikiranJ** [comment]
I want to block api for specific user .

**Comment by Mause** [comment]
That library allows you to block specific users?

**Comment by SasikiranJ** [comment]
Ok thank you

**Comment by coder3112** [comment]
Please close this issue if it has been solved

**Comment by SasikiranJ** [comment]
Ok

**Comment by tiangolo** [comment]
Thanks for the help everyone! ☕ 

And thanks for closing the issue @SasikiranJ

> Sorry for the long delay! 🙈  I wanted to personally address each issue/PR and they piled up through time, but now I'm checking each one in order.
